examples/go/documentation/resources.go

```diff
@@ -0,0 +1,120 @@
+package main
+
+// DOCS_BLOCK_START:resources-define
+import (
+	"fmt"
+
+	"github.com/aws/constructs-go/constructs/v10"
+	"github.com/aws/jsii-runtime-go"
+	"github.com/hashicorp/terraform-cdk-go/cdktf"
+	"github.com/hashicorp/terraform-cdk/examples/go/documentation/generated/hashicorp/aws/dataawsregion"
+	"github.com/hashicorp/terraform-cdk/examples/go/documentation/generated/hashicorp/aws/dynamodbtable"
+	aws "github.com/hashicorp/terraform-cdk/examples/go/documentation/generated/hashicorp/aws/provider"
+
+	// DOCS_BLOCK_END:resources-define
+	"github.com/hashicorp/terraform-cdk/examples/go/documentation/generated/hashicorp/aws/securitygroup"
+	"github.com/hashicorp/terraform-cdk/examples/go/documentation/generated/hashicorp/aws/snstopic"
+	// DOCS_BLOCK_START:resources-define
+)
+
+func NewResourcesStack(scope constructs.Construct, name string) cdktf.TerraformStack {
+	stack := cdktf.NewTerraformStack(scope, &name)
+
+	aws.NewAwsProvider(stack, jsii.String("aws"), &aws.AwsProviderConfig{
+		Region: jsii.String("eu-central-1"),
+	})
+
+	region := dataawsregion.NewDataAwsRegion(stack, jsii.String("region"), &dataawsregion.DataAwsRegionConfig{})
+
+	dynamodbtable.NewDynamodbTable(stack, jsii.String("first-table"), &dynamodbtable.DynamodbTableConfig{
+		Name:    jsii.String(fmt.Sprintf("my-first-table-%s", *region.Name())),
+		HashKey: jsii.String("id"),
+		Attribute: []map[string]string{
+			{"name": "id", "type": "S"},
+		},
+		BillingMode: jsii.String("PAY_PER_REQUEST"),
+	})
+
+	// DOCS_BLOCK_END:resources-define
+	// DOCS_BLOCK_START:resources-escape-hatch
+	tableName := "my-second-table"
+	table := dynamodbtable.NewDynamodbTable(stack, jsii.String("second-table"), &dynamodbtable.DynamodbTableConfig{
+		Name:    &tableName,
+		HashKey: jsii.String("id"),
+		Attribute: []map[string]string{
+			{"name": "id", "type": "S"},
+		},
+	})
+	table.AddOverride(jsii.String("provisioner"), []map[string]map[string]string{
+		{"local-exec": {"command": fmt.Sprintf("aws dynamodb create-backup --table-name %s --backup-name %s-backup", tableName, tableName)}},
```

Comment:
Can we force go-fmt to format this on multiple lines? It'll be hard to read otherwise in the example.

Reply:
Yeah, we can by adding newlines ourselves and then formatting (it works different than Prettier)

Reply:
fixed via #2587